File: pygameGUI/bullet.py
import pygame



class Bullet():
    def __init__(self, position):
        self.position = position
        self.velocity = 30
        self.acceleration = -1
        self.scale = (200, 200)
        
        self.damage = 2

    
    def move(self):
        self.velocity += self.acceleration
        self.position = [self.position[0]+self.velocity, self.position[1]]

    
    def draw(self, screen):
        image = pygame.image.load("data/models/items/bullet.png")
        image = pygame.transform.scale(image, self.scale)
        imagerect = image.get_rect().move(self.position)
        self.imagerect = imagerect

        screen.blit(image, imagerect)

File: webGUI/static/bullet.js


export class Bullet {
    constructor(position) {
        this.position = position
        this.damage = 2
        this.velocity = 15
        this.acceleration = -0.14

        this.src = "models/items/bullet.png"
        this.img = this.draw();
    }

    accelerate() {
        this.velocity += this.acceleration
    }


    draw() {
        let img = document.createElement("img");
        img.src = this.src;
        document.getElementById("body").appendChild(img)
        img.style.position = "absolute";
        img.style.width = "10%";
        img.style.height = "auto";
        img.style.left = this.position[0];
        img.style.bottom = this.position[1];
        img.style.imageRendering= "pixelated";
        img.style.imageRendering= "-moz-crisp-edges";
        img.style.imageRendering= "crisp-edges";
        return img
    }


    getCurrentLeft() {
        let leftString = this.img.style.left;
        let left = parseFloat(leftString.replace("px", "").replace("%", ""))
        return left
    }
}